### PR TITLE
[Compute] Updating sdk to the new version And adding TierOption

### PR DESCRIPTION
**Pull request by haagha**
## Description

Updating SDK to use Compute 2024-03-01 api version and Disk 2023-10-02 API version.
Also adding TierOption parameter that is part of the 2023-10-02 Disk release.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by vidai-msft** [comment]
@haagha When this PR is ready for us to review, please assign it to me. Thanks!

File: src/Compute/Compute.Management.Sdk/README.md [hunk 1]
```diff
@@ -48,29 +48,29 @@ directive:
 
 ### 
 ``` yaml
-commit: f715b7fee5e648d06b17467b08473f6cbeee84e0
+commit: 5312cc6da7d6c051e0e96902f3718a9fb3e9e021
```
Comment (Sandido) [inline src/Compute/Compute.Management.Sdk/README.md:51]: make a comment in Changelog, src/Compute/Compute/ChangeLog.md

File: src/Compute/Compute.Management.Sdk/Generated/AvailabilitySetsOperations.cs [hunk 2]
```diff
@@ -105,7 +105,7 @@ public async Task<AzureOperationResponse<AvailabilitySet>> CreateOrUpdateWithHtt
             {
                 throw new ValidationException(ValidationRules.CannotBeNull, "this.Client.SubscriptionId");
             }
-            string apiVersion = "2023-09-01";
```
Comment (Sandido) [inline src/Compute/Compute.Management.Sdk/Generated/AvailabilitySetsOperations.cs:108]: write a PR title and description

File: src/Compute/Compute.Management.Sdk/README.md [hunk 3]
```diff
@@ -48,29 +48,29 @@ directive:
...
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/image.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/restorePoint.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/capacityReservation.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/logAnalytic.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/runCommand.json 
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2023-10-02/diskRPCommon.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2023-10-02/disk.json
```
Comment (Sandido) [inline src/Compute/Compute.Management.Sdk/README.md:69]: so this is for 2 api versions, gotchya.

File: src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs [hunk 4]
```diff
@@ -192,6 +192,13 @@ public partial class NewAzureRmSnapshotConfigCommand : Microsoft.Azure.Commands.
             HelpMessage = "Required if createOption is CopyFromSanSnapshot. This is the ARM id of the source elastic san volume snapshot.")]
         public string ElasticSanResourceId { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "When createOption CopyStart, the snapshot will be copied at a quicker speed. Possible values include: 'None', 'Enhanced'")]
+        [PSArgumentCompleter("Enhanced", "None")]
+        public string TierOption { get; set; }
```
Comment (Sandido) [inline src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs:200]: IS this supposed to be here? IS there a test for this new feature?

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs (hunk 4 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs:200]: yeah this seems new parameter to a cmdlet

**Comment by grizzlytheodore** [comment]
no build error, should be good to go

File: src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs (hunk 4 above)
Comment (haagha) [reply src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs:200]: The parameter was requested a while back and seemed to have been blocked by SDK release. Added that in officially as part of this PR.

**Comment by haagha** [comment]
@vidai-msft comments have been addressed, please merge. Thanks.

**Review by vidai-msft — APPROVED** [review]

**Review by Sandido — APPROVED** [review]

**Pull request merged**